frontend/packages/process-editor/src/hooks/useBpmnEditor.test.tsx

```diff
@@ -1,168 +1,259 @@
+  bpmnXml: undefined,
+};
 const layoutSetId = 'someLayoutSetId';
-const layoutSetsMock: LayoutSets = {
+const layoutSets: LayoutSets = {
   sets: [
     {
       id: layoutSetId,
       tasks: [mockBpmnDetails.id],
     },
   ],
 };
+const defaultBpmnApiContextProps: BpmnApiContextProps = {
+  availableDataTypeIds: [],
+  availableDataModelIds: [],
+  allDataModelIds: [],
+  layoutSets,
+  pendingApiOperations: false,
+  existingCustomReceiptLayoutSetId: undefined,
+  addLayoutSet: jest.fn(),
+  deleteLayoutSet: jest.fn(),
+  mutateLayoutSetId: jest.fn(),
+  mutateDataTypes: jest.fn(),
+  saveBpmn: jest.fn(),
+  openPolicyEditor: jest.fn(),
+  onProcessTaskAdd: jest.fn(),
+  onProcessTaskRemove: jest.fn(),
+};
+const taskType: BpmnTaskType = 'data';
+const extensionElements: BpmnExtensionElementsEditor = {
+  values: [
+    {
+      $type: 'altinn:TaskExtension',
+      taskType,
+    },
+  ],
+};
+const businessObject: BpmnBusinessObjectEditor = {
+  $type: BpmnTypeEnum.Task,
+  id: 'test',
+  extensionElements,
+};
+const element: TaskEvent['element'] = {
+  id: 'test',
+  businessObject,
+};
+const xml = '<testxml></testxml>';
 
-class BpmnModelerMockImpl {
-  public readonly _currentEventName: string;
-  public readonly _currentEvent: any;
-  private readonly eventBus: any;
-
-  constructor(currentEventName: string, currentEvent) {
-    this._currentEventName = currentEventName;
-    this._currentEvent = currentEvent;
-    this.eventBus = {
-      _currentEventName: this._currentEventName,
-      on: this.on,
-    };
-  }
-
-  on(eventName: string, listener: (event: any) => void) {
-    if (eventName === this._currentEventName) {
-      listener(this._currentEvent);
-    }
-  }
-
-  get(elementName: string) {
-    if (elementName === 'eventBus') {
-      return this.eventBus;
-    }
-  }
-}
-
-jest.mock('../utils/bpmnObjectBuilders', () => ({
-  getBpmnEditorDetailsFromBusinessObject: jest.fn().mockReturnValue({}),
-}));
-
-jest.mock('../contexts/BpmnConfigPanelContext', () => ({
-  useBpmnConfigPanelFormContext: jest.fn(() => ({
-    metadataFormRef: { current: null },
-    resetForm: jest.fn(),
-  })),
-}));
+// Mocks:
+jest.mock('bpmn-js/lib/Modeler', () => jest.fn().mockImplementation(bpmnModelerImplementation));
 
-jest.mock('../contexts/BpmnContext', () => ({
-  ...jest.requireActual('../contexts/BpmnContext'),
-  useBpmnContext: jest.fn(() => ({
-    getUpdatedXml: jest.fn(),
-    modelerRef: { current: null },
-    setBpmnDetails: setBpmnDetailsMock,
-  })),
-}));
+function bpmnModelerImplementation(): BpmnModeler {
+  return {
+    get: getModeler,
+    importXML: () => Promise.resolve({ warnings: [] }),
+    on: <K extends keyof EventMap>(eventName: K, callback: EventMap[K]) => {
+      return eventListeners.add(eventName, callback);
+    },
+    off: <K extends keyof EventMap>(eventName: K, callback: EventMap[K]) => {
+      return eventListeners.remove(eventName, callback);
+    },
+    saveXML: () => Promise.resolve({ xml }),
```

Comment:
Nitpicking; Can these be placed in separate consts as well?

Reply:
Sure. @coderabbitai, can you create a committable suggestion that fixes this?

Reply:
Not exactly what I expected. I'll do it myself.